unity-renderer/Assets/DCLPlugins/ECS7/Utils/Scene/Bounds.cs

```diff
@@ -0,0 +1,106 @@
+using DCL.Configuration;
+using DCL.Helpers;
+using System.Collections.Generic;
+using UnityEngine;
+
+/*
+* Scene utils file for scene bounds related stuff
+*/
+
+public static partial class UtilsScene
+{
+    public static Bounds CalculateOuterBounds(Vector2Int[] parcels, Vector3 baseParcelWorldPos)
+    {
+        Bounds outerBounds = new Bounds();
+        outerBounds.SetMinMax(new Vector3(baseParcelWorldPos.x, 0f, baseParcelWorldPos.z),
+            new Vector3(baseParcelWorldPos.x + ParcelSettings.PARCEL_SIZE, 0f, baseParcelWorldPos.z + ParcelSettings.PARCEL_SIZE));
+
+        for (int i = 0; i < parcels.Length; i++)
+        {
+            // Update outer bounds with parcel's size
+            var parcel = parcels[i];
+
+            Vector3 parcelWorldPos = PositionUtils.WorldToUnityPosition(Utils.GridToWorldPosition(parcel.x, parcel.y));
+            outerBounds.Encapsulate(new Vector3(parcelWorldPos.x, 0, parcelWorldPos.z));
+            outerBounds.Encapsulate(new Vector3(parcelWorldPos.x + ParcelSettings.PARCEL_SIZE, 0, parcelWorldPos.z + ParcelSettings.PARCEL_SIZE));
+        }
+
+        // Apply outer bounds extra threshold
+        outerBounds.SetMinMax(new Vector3(outerBounds.min.x - ParcelSettings.PARCEL_BOUNDARIES_THRESHOLD, 0f, outerBounds.min.z - ParcelSettings.PARCEL_BOUNDARIES_THRESHOLD),
+            new Vector3(outerBounds.max.x + ParcelSettings.PARCEL_BOUNDARIES_THRESHOLD, 0f, outerBounds.max.z + ParcelSettings.PARCEL_BOUNDARIES_THRESHOLD));
+
+        return outerBounds;
+    }
+
+    public static bool IsInsideSceneInnerBounds(ISet<Vector2Int> sceneParcels, float sceneHeightLimit, Vector3 targetWorldPosition, float height = 0f)
```

Comment:
```suggestion
    public static bool IsInsideSceneInnerBounds(IReadOnlyCollection<Vector2Int> sceneParcels, float sceneHeightLimit, Vector3 targetWorldPosition, float height = 0f)
```

Reply:
Better to leave HashSet as ICollection or other interfaces either don't have `Contains()` or have to be traversed by a foreach/iterator